### PR TITLE
Remove lazy effect and model mapping from OneWaySeq binding

### PR DESCRIPTION
**Pull request by TysonMN**

### CODE REVIEW

**Comment by TysonMN** [comment]
@[marner2](https://github.com/marner2), I tried to add you as a reviewer, but GitHub won't allow it.

File: src/Elmish.WPF/Binding.fs
Comment (TysonMN) [inline src/Elmish.WPF/Binding.fs:501]: I should a add this (beta-breaking) change to the release notes

File: src/Elmish.WPF/BindingInternal.fs
Comment (the-not-mad-psychologist) [inline src/Elmish.WPF/BindingInternal.fs:407]: Shouldn't `outMapA` and `inMapA` be renamed to `outMap` and `inMap` since they are no longer distinguished from other map functions?

**Review by the-not-mad-psychologist — APPROVED** [review]

File: src/Elmish.WPF/BindingInternal.fs
Comment (marner2) [reply src/Elmish.WPF/BindingInternal.fs:407]: `outMapId` needs to be distinguished

File: src/Elmish.WPF/Binding.fs
Comment (marner2) [reply src/Elmish.WPF/Binding.fs:501]: Was this the only place this was used? I thought there were more...

File: src/Elmish.WPF/BindingInternal.fs
Comment (marner2) [inline src/Elmish.WPF/BindingInternal.fs:173]: I thought we were going to have issues here, but yeah I do see now how this is fully redundant with `Get`

**Review by marner2 — APPROVED** [review]
I'm pretty sure this will conflict a bit with my existing stuff. I removed #471 since it conflicts heavily, but I'm also adding types here in #473

File: src/Elmish.WPF/Binding.fs
Comment (TysonMN) [reply src/Elmish.WPF/Binding.fs:501]: Oh, that is not part of the public API.  Instead, I should expose something like this to the public API.

File: src/Elmish.WPF.Tests/ViewModelTests.fs
Comment (TysonMN) [inline src/Elmish.WPF.Tests/ViewModelTests.fs:529]: Like the comment says, PR #478 needs to merge first so this test will pass

Comment (marner2) [reply src/Elmish.WPF.Tests/ViewModelTests.fs:529]: This makes sense. So basically, by adding the lazy effect, it was calling "Get" once for each side of the equality check. Afer PR #478, it has the getter right there to call (which doesn't affect this call).

**Review by marner2 — APPROVED** [review]
Approved for when the other gets merged in and everything passes.

**Comment by TysonMN** [comment]
Nice!  I can "Update Branch" (i.e. rebase the branch) from the GitHub app.  I had done a merge commit, so I will keep doing that.

File: src/Elmish.WPF.Tests/ViewModelTests.fs
Comment (TysonMN) [reply src/Elmish.WPF.Tests/ViewModelTests.fs:530]: > [...] by adding the lazy effect, it was calling "Get" once for each side of the equality check. Afer PR #478, it has the getter right there to call (which doesn't affect this call).

Yes, the `Get` was composed into both "sides" of the equality.  That is where two calls came from. The third call was in the code that prepares to call merge.  This is because the model mapping in the Lazy case mapped both the equals function and the nested binding.

Now this third call won't happen because the model that the merge preparation code needs is now passed down to it from the Lazy case.

**Comment by marner2** [comment]
@TysonMN You waiting on something to merge this?

**Comment by TysonMN** [comment]
Just the time in the day to click the button

**Pull request merged**
